### ISSUE TITLE
[FEATURE] Tumor mutational burden (TMB) scoring

### ISSUE THREAD

**Issue by apeltzer** [nf-core/sarek#495]
# nf-core/sarek feature request

## Is your feature request related to a problem? Please describe

TMB / tumor mutational burden is a key metric for many oncology related projects and thus could be a feature requested by many users of the pipeline at some point to be computed similar to MSI status (which is already done in the pipeline). 

## Describe the solution you'd like

Have some method / way of computing TMB status and reporting this to the user in the main MultiQC table ideally.

## Describe alternatives you've considered

![image](https://user-images.githubusercontent.com/2359510/157202429-9ecbbc57-5ffc-4c87-b8e8-5612b3a0bc9e.png)

**Comment by apeltzer** [comment]
Easy to use feature would be this here: https://github.com/bioinfo-pf-curie/TMB

**Comment by apeltzer** [comment]
A good explanation how to do the job potentially as this evaluates several panels + WES that could be of interest: https://www.nature.com/articles/s41598-020-68394-4

**Comment by apeltzer** [comment]
potentially worth looking at too: https://github.com/bcgsc/TMBur

**Comment by apeltzer** [comment]
Also a good tool apparently, estimating TMB: https://github.com/bioinform/ecTMB

**Comment by apeltzer** [comment]
Strong tendency towards using the curie approach because:

- conda recipe --> biocontainer in place ✔️ 
- good examples / documentation in place ✔️ 
- straightforward output for multiqc (tsv tables are produced automatically --> could just report this in separate table in MultiQC report in Sarek) ✔️ 

All these points above are sometimes hard to achieve if you have to do them yourself --> good idea to rely on this instead 👍🏻

**Comment by maxulysse** [comment]
>     * straightforward output for multiqc (tsv tables are produced automatically --> could just report this in separate table in MultiQC report in Sarek) heavy_check_mark

MultiQC module possibility even?

**Comment by apeltzer** [comment]
No, nothing else - I suspect that they have their own conda channel where they provide everything this uses / needs.

**Comment by apeltzer** [comment]
Hey everyone, I've just made a bioconda recipe for the curie tmb tool --> see here: https://github.com/bioconda/bioconda-recipes/pull/35393 

@nservant you could also link to that recipe in the future, as it would allow people to directly install the tool without having to setup a separate environment, e.g.:

`mamba install -c bioconda -c conda-forge tmb=1.3.0`

Running  the tools then works with: 

`pyTMB.py <parameters` (and the genome size script works the same way)

**Comment by FriederikeHanssen** [comment]
❤️ thank you for adding this! As soon as biocontainers is there, I'll add the module.

**Comment by FriederikeHanssen** [comment]
> Hello @FriederikeHanssen @apeltzer,
> 
> I am the co-developper of pyTMB, I'm working with Nicolas Servant.
> 
> Thanks a lot for your help for bioconda ! I was struggling with the build.sh part.
> 
> Let me know If I can help for the config files or for testing

Hi @tomgutman !

This PR has been a long time in the making due to me no t having time, but also because creating the yml files seems very daunting. Do you already have some for VEP or any of the variantcallers that we use, such as strelka, and freebayes?

**Comment by Raghu9721** [comment]
Hello @apeltzer @FriederikeHanssen 

This is Raghavendra,
I am currently using pyTMB.py for finding Tumor mutational burden with having VEP annotated vcf output file. But I am unable to use pyTMB.py due to non-availability of specific config.yml file for VEP. Please help me in creating it.

**Comment by FriederikeHanssen** [comment]
Hi @Raghu9721 ! Sorry just saw this, I am not the pyTMB tool developer. I would ask here to get help from them for this: https://github.com/bioinfo-pf-curie/TMB

**Comment by nservant** [comment]
Hi @Raghu9721, @FriederikeHanssen , Happy to help on that question, but I never used VEP before. The only thing I would need is a VCF file annotated with VEP to see how the different fields can be parsed.

**Comment by bounlu** [comment]
I am curious to see whether there is any progress to add TMB to the sarek pipeline.

**Comment by GCAT01** [comment]
I see the score check above was closed, was this ever implemented? I work in a clinical setting and calculating TMB is a big part of our workflow.

**Comment by RSWilson1** [comment]
I saw this PR was merged, is there a plan to add this to the next version of Sarek?
